app/controllers/journeys/expenses/goodsToSellOrUse/DisallowableGoodsToSellOrUseAmountController.scala

```diff
@@ -35,7 +35,7 @@ import scala.concurrent.{ExecutionContext, Future}
 
 @Singleton
 class DisallowableGoodsToSellOrUseAmountController @Inject() (override val messagesApi: MessagesApi,
-                                                              sessionRepository: SessionRepository,
+                                                              selfEmploymentService: SelfEmploymentService,
```

Comment:
It may be better to inject the abstraction rather than the impl itself, as it gives us more flexibility to swap out/test.

Reply:
Happy for me to do this in a new PR where I switch it everywhere to the BaseService?

Reply:
Yep no probs